src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/AggregationManager.cs

```diff
@@ -0,0 +1,314 @@
+                    cancelToken.WaitHandle.WaitOne(delayTime);
+
+                    // don't do collection if timer may not have run to completion
+                    if (cancelToken.IsCancellationRequested)
+                    {
+                        break;
+                    }
+
+                    // collect statistics for the completed interval
+                    _beginCollection(intervalStartTime);
+                    Collect();
+                    _endCollection(intervalStartTime);
+                    intervalStartTime = nextIntervalStartTime;
+                }
+            }
+            catch (Exception e)
+            {
+                _collectionError(e);
+            }
+        }
+
+        public void Dispose()
+        {
+            _cts.Cancel();
+            if (_collectThread != null)
+            {
+                _collectThread.Join();
+                _collectThread = null;
+            }
+            _listener.Dispose();
+        }
+
+        private void RemoveInstrumentState(Instrument instrument, InstrumentState state)
+        {
+            _instrumentStates.TryRemove(instrument, out InstrumentState? _state);
+        }
+
+        private InstrumentState? GetInstrumentState(Instrument instrument)
+        {
+            if (!_instrumentStates.TryGetValue(instrument, out InstrumentState? instrumentState))
+            {
+                lock (this) // protect _isntrumentConfigFuncs list
+                {
+                    foreach (var filter in _instrumentConfigFuncs)
+                    {
+                        if (filter(instrument))
+                        {
+                            instrumentState = BuildInstrumentState(instrument);
+                            if (instrumentState != null)
+                            {
+                                _instrumentStates.TryAdd(instrument, instrumentState);
+                                instrumentState = _instrumentStates[instrument];
+                            }
+                            break;
+                        }
+                    }
+                }
+            }
+            return instrumentState;
+        }
+
+        internal InstrumentState? BuildInstrumentState(Instrument instrument)
+        {
+            Func<Aggregator>? createAggregatorFunc = GetAggregatorFactory(instrument);
+            if (createAggregatorFunc == null)
+            {
+                return null;
+            }
+            Type aggregatorType = createAggregatorFunc.GetType().GenericTypeArguments[0];
+            Type instrumentStateType = typeof(InstrumentState<>).MakeGenericType(aggregatorType);
+            return (InstrumentState)Activator.CreateInstance(instrumentStateType, createAggregatorFunc)!;
+        }
+
+        private static Func<Aggregator>? GetAggregatorFactory(Instrument instrument)
+        {
+            Type type = instrument.GetType();
+            Type? genericDefType = null;
+#if IS_GENERIC_TYPE_SUPPORT
```

Comment:
Why is this #if needed?

Reply:
While I was doing development I think a concurrent change came in that removed the support for one of the old targets this library supported (netstandard1.1 perhaps?). The rebase eliminated the define in the project but I had forgotten to remove the usage in the code.